app/flatpak-builtins-build.c

```diff
@@ -504,12 +505,21 @@ flatpak_builtin_build (int argc, char **argv, GCancellable *cancellable, GError
 
   flatpak_run_apply_env_vars (bwrap, app_context);
 
+  if (is_app)
+    {
+      /* We don't actually know the final branchname yet, so use "nobranch" as fallback to avoid unexpected matches.
+         This means any extension point used at build time must have explicit versions to work. */
+      g_autofree char *fake_ref = g_strdup_printf ("app/%s/%s/nobranch", id, runtime_ref_parts[2]);
+      if (!flatpak_run_add_extension_args (bwrap, metakey, fake_ref, FALSE, &app_extensions, cancellable, error))
+        return FALSE;
```

Comment:
Seems suboptimal. What is the reason why we can't move this closer to where we know the branchname? Or could we provide it on `build-init`?

Reply:
The branch is currently only available in build-export, and changing that is a massive change. However, its not really an issue in practice. The extension your build depends on should not depend on the version of the app you're building, you need to specify a specific version of the extension you want.